DetectDynamicJS.py

```diff
@@ -81,14 +81,17 @@ def doPassiveScan(self, baseRequestResponse):
             requestHeaders = request.tostring()[:requestBodyOffset].split('\r\n')
             requestBody = request.tostring()[requestBodyOffset:]
             modified_headers = "\n".join(header for header in requestHeaders if "Cookie" not in header)
-            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers+requestBody))
+            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers + requestBody))
+            newResponseInfo = self._helpers.analyzeResponse(newResponse.getResponse())
+            if not (self.hasValidStatusCode(newResponseInfo.getStatusCode()) and self.hasBody(newResponseInfo.getHeaders())):
```

Comment:
The `newResponse` can be of any status and doesn't need a body. If the authenticated version is a script, which we know, and this isn't, we know there is going to be at least a login oracle. If the authenticated version includes interesting data is unknown. Returning `None` will lose that information.